shell/edgexfoundry-snapcraft.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash -x
+
+# note that since this script gets included into a yaml string using jjb's
+# include-raw-escape macro, we can't use any squiggly braces here, otherwise
+# they will get duplicated
+# for some reason we also can't use include-raw macro because that macro will
+# attempt to interpret anything contianed in a macro as a jenkins key to
+# expand which we also don't want
+
+# if we are running inside a jenkins instance then copy the login file 
+# and check if this is a release job
+if [ -n "$JENKINS_URL" ]; then
+    # since we are running from jenkins, use the workspace as the git root
+    GIT_ROOT=$WORKSPACE
+    if [ -f "$HOME/EdgeX" ]; then
+       cp "$HOME/EdgeX" "$GIT_ROOT/edgex-snap-store-login"
+    else
+        echo "I seem to be running on Jenkins, but there's not a snap store login file..." 
+    fi
+
+    # figure out what kind of job this is using $JOB_NAME and simplify that 
+    # into $JOB_TYPE
+    if [[ "$JOB_NAME" =~ .*-snap-.*-stage-snap.* ]]; then
+        JOB_TYPE="stage"
+    elif [[ "$JOB_NAME" =~ .*-snap-release-snap ]]; then
+        JOB_TYPE="release"
+    else
+        JOB_TYPE="build"
+    fi
+else
+    # not running in a jenkins container, assume the current working directory
+    # is the repo git dir that we want
+    GIT_ROOT=$PWD
+fi
+
+# find the name of this snap from the snapcraft.yaml "name" key
+# note that we assume that snapcraf.yaml is located in snap/snapcraft.yaml 
+# but it is technically allowed to store it in other locations, we just don't
+# support that here
+# TODO: implement better handling of snapcraft.yaml file discovery
+SNAP_NAME=$(grep -Po '^name: \K(.*)' snap/snapcraft.yaml)
+
+# save the entrypoint.sh script inside the build context we will send to docker
```

Comment:
Another sentence that seems to have an implicit end at EOL.

Reply:
Fixed